### PR TITLE
Added primitive visitor to serialize radio packets

### LINKED ISSUE #235: Create a visitor to create radio packets from Primitives
**Is your feature request or enhancement related to a problem? What specifically could be better? Please describe.**
We should use a visitor to create radio packets from the `Primitive` classes to remove the responsibility from the `Primitive` classes.

**Describe the solution you'd like**
Create a new `RadioSerializerPrimitiveVisitor` that extends from the `PrimitiveVisitor` interface. This visitor should be responsible for creating a radio packet from the information in each Primitive. This includes packing the parameters into the parameters and bits, as well as the `Primitive Type` id (this will be handled by the visitor instead of #221 )

**Describe alternatives you've considered**
Leaving things as they are. This means that the `Primitive` classes are partially aware and in-control of the radio encoding, because they must return the parameters and "extra bits" in the correct order from the `getParameters` and `getExtraBits` functions. This is not ideal.

**Additional context**
A similar approach is being taken to use a Visitor to encode the `Primitives` to ROS messages. See #226

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description
*Give a high-level description of the changes in this PR*
Added a new `PrimitiveVisitor` that will be responsible for serializing `Primitive` classes into radio packets.

Note: I've linked issue #63 for all the `TODOs`, as there's not a specific issue for radio serialization yet. I'll leave this up to @henrymliu as he has a better idea of the scope of radio changes and how they're being broken up. Similarly, the `uint64_t` return type for the radio packet is mostly a placeholder as I'm not sure exactly what time will be returned. I'll also leave this up to @henrymliu to correct when the implementation is being implemented.

I'm also hoping #126 and #151 get merged first so they don't need to be updated. That means this PR will need to be updated, but it will simply be adding new `visit` functions.

### Testing Done
*Outline any testing that was done for these changes. This could be unit tests, integration tests, etc.*
None, as there is no implementation.

### Resolved Issues
*Link any issues that this PR resolved. Eg `resolves #1, #2, and #5` (note that they MUST be specified like this so Github can automatically close them then this PR merges)*
resolves #235 

### Length Justification

N/A

### Review Checklist
*(Please check every item to indicate your code complies with it (by changing `[ ]`->`[x]`). This will hopefully save both you and the reviewer(s) a lot of time!)*
***It is the reviewers responsibility to also make sure every item here has been covered***
- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom` 
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

***Feel free to make additions of things that we should be checking to this file if you think there's something missing!!!!***

### CODE REVIEW

File: src/thunderbots/software/radio_communication/visitor/radio_packet_serializer_primitive_visitor.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+     */
+    void visit(const PivotPrimitive &pivot_primtiive) override;
+
+    /**
+     * Returns the most recent serialized packet created by this
+     * RadioPacketSerializerPrimitiveVisitor. This is the radio packet created by
+     * one of the 'visit' functions
+     *
+     * Calling this function before this Visitor has been called at least once has
+     * undefined behavior
```
Comment (garethellis0) [inline src/thunderbots/software/radio_communication/visitor/radio_packet_serializer_primitive_visitor.h:65]: :nauseated_face: could we return an optional instead?

**Review by garethellis0 — COMMENTED** [review]
I'm fine with merging this, but seems a tad pointless? Might as well have just passed this over to Henry and got him to fill it in and open a PR?

File: src/thunderbots/software/radio_communication/visitor/radio_packet_serializer_primitive_visitor.h (hunk 1 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/radio_communication/visitor/radio_packet_serializer_primitive_visitor.h:65]: Yep. I'll change this for the `GrSimCommandPrimitiveVisitor` too

**Comment by MathewMacDougall** [comment]
True, but I'm assuming with the implementation it would be a large PR. Figured we could break it up.

**Comment by MathewMacDougall** [comment]
@henrymliu Would you rather just incorporate this directly with your work? I have no strong preference, and that would introduce a natural delay to allow the last Primitive PRs to be merged first.

**Comment by henryymliu** [comment]
I think I would rather integrate this with my current work, would make things easier to test as well.

**Comment by MathewMacDougall** [comment]
Ok. I'll close this PR then but you can still use it as a reference?

**Comment by henryymliu** [comment]
Yes that sounds good!

**Pull request closed without merging**